Support job cancellation through details panel

jakemcdermott opened the pull request:
##### SUMMARY
for #7657 

edit: also addresses #8838 cc @nixocio

jakemcdermott committed:
Support job cancellation through details panel

nixocio commented on awx/ui_next/src/screens/Job/JobDetail/JobDetail.jsx:
```diff
@@ -58,6 +60,16 @@ const VERBOSITY = {
+const getJobModel = type => {
+  if (type === 'ad_hoc_command') return AdHocCommandsAPI;
+  if (type === 'inventory_update') return InventoriesAPI;
+  if (type === 'project_update') return ProjectUpdatesAPI;
```
hey @jakemcdermott, I got this code locally, and when I attempted to `cancel` a project update on the job details page, and I got the following error. The same error does not appear if I cancel the job on the job output.  

https://user-images.githubusercontent.com/9053044/112148245-09387080-8bb4-11eb-9443-37dfac89e63a.mov

jakemcdermott replied:
@nixocio 

:facepalm: Looks like the `cancel` methods aren't actually defined on all of the models yet. I'll fix this shortly.

jakemcdermott replied:
https://github.com/ansible/awx/pull/9659/commits/72d700e12587b3dacba79c91b983ce5a6fac5c5f

jakemcdermott committed:
Use the type-specific job models

Add the expected launch and run methods to the different job type
models. Include a new helper function to look up the right model
given a job type and use it in place of switch statements or
passing the type in to build a url.

jakemcdermott commented:
note: [e2e currently looks ok for this pr](https://dashboard.cypress.io/projects/ui2mq7/runs/5018/overview). I see two failing tests, but I see the same two tests failing on devel.
